File: AI_Crop_Disease_Detection_System/AI Backend/app/Models/Disease.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Disease extends Model
{
    protected $fillable = [
        'crop_id',
        'name',
        'name_sn',
        'class_label',
        'description',
        'description_sn',
        'symptoms',
        'symptoms_sn',
        'prevention',
        'prevention_sn',
        'is_active',
    ];

    protected function casts(): array
    {
        return [
            'is_active' => 'boolean',
        ];
    }

    public function crop(): BelongsTo
    {
        return $this->belongsTo(Crop::class);
    }

    public function treatments(): HasMany
    {
        return $this->hasMany(Treatment::class);
    }

    public function diagnoses(): HasMany
    {
        return $this->hasMany(Diagnosis::class);
    }
}

File: AI_Crop_Disease_Detection_System/AI Frontend/lib/src/models/disease.dart
import 'treatment.dart';

class Disease {
  const Disease({
    required this.id,
    required this.name,
    required this.classLabel,
    this.description,
    this.symptoms,
    this.prevention,
    this.treatments = const [],
  });

  final int id;
  final String name;
  final String classLabel;
  final String? description;
  final String? symptoms;
  final String? prevention;
  final List<Treatment> treatments;

  factory Disease.fromJson(Map<String, dynamic> json) {
    final treatmentsJson = json['treatments'] as List<dynamic>? ?? [];

    return Disease(
      id: json['id'] as int,
      name: json['name'] as String,
      classLabel: json['class_label'] as String,
      description: json['description'] as String?,
      symptoms: json['symptoms'] as String?,
      prevention: json['prevention'] as String?,
      treatments: treatmentsJson
          .map((treatment) => Treatment.fromJson(treatment as Map<String, dynamic>))
          .toList(),
    );
  }
}
